### DISCUSSION TITLE
Portfolio as subset of stocks data

### DISCUSSION THREAD

**Discussion by FabioMD1972** [AI4Finance-Foundation/FinRL#1145] [Q&A]
Hi,
I'm quite new on FinRL.
This is my usecase:

- suppose to collect DOW 30 data and use it to build the environment
- within this environment I want to trade only on a (fixed) subset of ticks lets says [AAPL, AXP]

The question is how I have to instantiate the StockTradingEnv class?

`
from finrl.meta.env_stock_trading.env_stocktrading import StockTradingEnv
....

env_kwargs = {
    "hmax": 100,
    "initial_amount": 1000000,
    "num_stock_shares": num_stock_shares,
    "buy_cost_pct": buy_cost_list,
    "sell_cost_pct": sell_cost_list,
    "state_space": state_space,
    "stock_dim": stock_dimension,
    "tech_indicator_list": INDICATORS,
    "action_space": stock_dimension,
    "reward_scaling": 1e-4
}

e_train_gym = StockTradingEnv(df = train, **env_kwargs)
`

I look at a lot of examples but I can't find the solution.

Thank you, F.

**Comment by devamin** [comment]
Just filter your train dataframe just after you read it before calculating any dimensions, keep only the subset of ticks you want.
